agent/consul/leader_connect.go

```diff
@@ -111,6 +123,54 @@ func (s *Server) pruneCARoots() error {
 	return err
 }
 
+func (s *Server) runVirtualIPVersionCheck(ctx context.Context) error {
+	// Return early if the flag is already set.
+	_, err := s.setVirtualIPVersionFlag()
+	if err != nil {
+		s.loggers.Named(logging.Connect).Warn("error enabling virtual IPs", "error", err)
+	}
```

Comment:
I think we are missing the return here when the first return value is `true`.
```suggestion
	// Return early if the flag is already set.
	done, err := s.setVirtualIPVersionFlag()
	if err != nil {
		s.loggers.Named(logging.Connect).Warn("error enabling virtual IPs", "error", err)
	}
	if done {
		return nil
	}
```

Does that seem right?

Reply:
For some reason, doing this causes the goroutine to leak (and TestAgentLeaks_Server to fail) - I looked at this for a bit but i'm not sure why.

Reply:
Ohh, right! I ran into the same problem when I tried to do this, but I also was not sure why.

I think I found a workaround, which was to call `leaderRoutineManager.Stop()`, but now I can't find the link to where I did that. It might be possible to call `Stop` here before returning.

Reply:
That worked; calling `leaderRoutineManager.Stop()` before returning was enough.